### PR TITLE
feat: allow contract to be paused and upgraded

### PR DESCRIPTION
**Pull request by raventid**

### CODE REVIEW

File: contract/src/lib.rs [hunk 1]
```diff
@@ -340,6 +392,7 @@ impl Contract {
     /// @param merkleProof  merkle tree path (concatenated LE sha256 hashes) (does not contain initial transaction_hash and merkle_root)
     /// @param confirmations how many confirmed blocks we want to have before the transaction is valid
     /// @return True if txid is at the claimed position in the block at the given blockheight, False otherwise
+    #[pause]
```
Comment (karim-en) [inline contract/src/lib.rs:395]: Let's add except
#[pause(except(roles(Role::UnrestrictedVerifyTransaction)))]

**Review by karim-en — CHANGES_REQUESTED** [review]

File: contract/src/lib.rs (hunk 1 above)
Comment (raventid) [reply contract/src/lib.rs]: Added a role

File: contract/src/lib.rs [hunk 2]
```diff
@@ -152,6 +203,7 @@ impl Contract {
     /// Saving block header received from a Bitcoin relay service
     /// This method is private but critically important for the overall execution flow
     #[handle_result]
+    #[pause]
```
[the remark this answers is not in the document]
Comment (raventid) [reply contract/src/lib.rs]: Added a role

File: contract/src/lib.rs (hunk 1 above)
Comment (karim-en) [reply contract/src/lib.rs]: Why the role is `UnrestrictedSubmitBlocks` instead of `UnrestrictedVerifyTransaction`?

Comment (raventid) [reply contract/src/lib.rs]: @karim-en Attention issue. Fixed in https://github.com/aurora-is-near/btc-light-client-contract/pull/8/commits/7bf503f0a537fc46a5a441cfaccc5ad6123f17d0

**Review by karim-en — APPROVED** [review]

**Pull request merged**